backend/api/EventHandlers/MissionEventHandler.cs

```diff
@@ -149,8 +149,24 @@ private async void OnRobotAvailable(object? sender, RobotAvailableEventArgs e)
             }
 
             MissionRun? missionRun;
-            if (robot.MissionQueueFrozen) { missionRun = await MissionService.ReadNextScheduledEmergencyMissionRun(robot.Id); }
-            else { missionRun = await MissionService.ReadNextScheduledMissionRun(robot.Id); }
+
+            missionRun = await MissionService.ReadNextScheduledLocalizationMissionRun(robot.Id);
+
+            if (missionRun != null)
+            {
+                _logger.LogInformation("The robot will be localized in the next mission");
+            }
+            else
+            {
+                if (robot.MissionQueueFrozen)
```

Comment:
Consider keeping it inline rather than splitting over multiple lines

Reply:
Improves readability but I will remove if you say so .

Reply:
done.